sled-agent/src/serial.rs

```diff
@@ -179,15 +180,20 @@ impl SerialConsoleBuffer {
                         }
                     }
                 });
+            // how long to sleep when nothing happens (to avoid thrashing CPU with polling here)
+            const SLEEP_DURATION: Duration = Duration::from_secs(1);
```

Comment:
Although this will mitigate the CPU thrashing, won't think result in an artificial one-second delay each time the serial console idles?

(Without trying anything myself, I would have figured that this would cause the console to appear laggy)

Reply:
yeah - as you suggested elsewhere, it might be a better idea to simply break from the loop in some of these cases